solidity/dashboard/src/components/AddTopUpModal.jsx

```diff
@@ -1,39 +1,209 @@
+import { withConfirmationModal } from "./ConfirmationModal"
+import Button from "./Button"
+import Divider from "./Divider"
+import TokenAmount from "./TokenAmount"
+import { KEEP } from "../utils/token.utils"
+import { add } from "../utils/arithmetics.utils"
+import { useModal } from "../hooks/useModal"
 
 const AddTopUpModal = ({
   authorizerAddress,
   beneficiary,
   operatorAddress,
-  ...formProps
+  currentAmount,
+  onBtnClick: submitConfirmationModal,
+  onCancel,
+  availableAmount,
+}) => {
+  const [step, setStep] = useState(1)
+  const [amount, setAmount] = useState("0")
+
+  const onSubmit = (values) => {
+    if (step === 1) {
+      setAmount(values.amount)
+      setStep((preveStep) => preveStep + 1)
+    } else if (step === 2) {
+      submitConfirmationModal({ amount })
+    }
+  }
+
+  return (
+    <>
+      {step === 1 && (
+        <Step1
+          authorizerAddress={authorizerAddress}
+          beneficiary={beneficiary}
+          operatorAddress={operatorAddress}
+          currentAmount={currentAmount}
+          onSubmit={onSubmit}
+          availableAmount={availableAmount}
+          onCancel={onCancel}
+        />
+      )}
+      {step === 2 && (
+        <Step2
+          title={`You are adding ${KEEP.toFormat(
+            amount
+          )} KEEP to this delegation. Note that this top up cannot be canceled once initiated.`}
+          confirmationText="CONFIRM"
+          getLabelText={(confirmationText) =>
+            `Type ${confirmationText} to add KEEP.`
+          }
+          btnText="confirm"
+          onBtnClick={onSubmit}
+          onCancel={onCancel}
+          authorizerAddress={authorizerAddress}
+          beneficiary={beneficiary}
+          operatorAddress={operatorAddress}
+          newAmount={add(KEEP.fromTokenUnit(amount), currentAmount)}
+          amountToAdd={amount}
+          onSubmit={onSubmit}
+        />
+      )}
+    </>
+  )
+}
+
+const Step1 = ({
+  authorizerAddress,
+  beneficiary,
+  operatorAddress,
+  currentAmount,
+  availableAmount,
+  onSubmit,
+  onCancel,
 }) => {
   return (
     <>
       <h3 className="mb-1">
         Enter an amount of KEEP to add to this existing delegation.
       </h3>
-      <span className="text-big text-grey-70">Delegation Details</span>
-      <DelegationAddress address={authorizerAddress} label="authorizer" />
-      <DelegationAddress address={operatorAddress} label="operator" />
-      <DelegationAddress address={beneficiary} label="beneficiary" />
-      <AmountForm {...formProps} />
+      <h4 className="text-grey-70 mb-1">Delegation Details</h4>
+      <DelegationDetails
+        authorizerAddress={authorizerAddress}
+        beneficiary={beneficiary}
+        operatorAddress={operatorAddress}
+      />
+      <div className="flex row center mt-2 mb-2">
+        <div className="flex-1">
+          <Tag text="Current" IconComponent={Icons.KeepToken} />
+        </div>
+        <TokenAmount amount={currentAmount} />
+      </div>
```

Comment:
![image](https://user-images.githubusercontent.com/40306834/116397655-269dd180-a827-11eb-8888-10bee03c67c9.png)

I'm thinking that we could also show the amount of KEEP placed in the top-ups for this address just below the amount of KEEP in the delegation. E.g right now this address has a delegation of a 60k KEEP but there is also a top-up for this address with the amount of 178k KEEP that is not commited. Might be a good idea to display this in this modal (and probably also in Step2 modal). Thoughts?

Reply:
Makes sense for me but not sure if we should add not committed top-ups to the current amount because not committed tokens do not participate in the work of the network. So I would suggest rather list pending top-ups but not sure how we can handle this in the UI, so I'll ping @liz-shinn here.

Also sounds like we should address it in a separate PR to not block the v.1.13.0 release. I assume that multiple top-ups for the same delegation are rather rare.

Reply:
>E.g right now this address has a delegation of a 60k KEEP but there is also a top-up for this address with the amount of 178k KEEP that is not commited. Might be a good idea to display this in this modal (and probably also in Step2 modal). Thoughts?

So this is a scenario in which the user has a previously submitted top up but has not committed it and wants to submit an additional top up, correct?

Also, I think we should display the full addresses here since there's room in the modal to do so versus abbreviating them.

Reply:
What about something like this @michalsmiarowski @r-czajkowski 
In figma here: https://www.figma.com/file/Su61MwHLrQn3ld24q85pMx/Token-Dashboard-dApp?node-id=8916%3A0

<img width="405" alt="Screen Shot 2021-04-29 at 11 33 08 AM" src="https://user-images.githubusercontent.com/57226633/116593762-af2b7900-a8de-11eb-93b3-0b0a0422918f.png">

Reply:
Idk if "top-up" or "top up" – whichever is consistent with how we've referred to it in the UI previously